src/cephadm/cephadm.py

```diff
@@ -6035,7 +6040,19 @@ def mgr_has_latest_epoch():
     return ctx.error_code
 
 ##################################
-
+def command_sudo_login(ctx: CephadmContext) -> int:
```

Comment:
NIT, we use the format `command_xxx` for func names for funcs that users can call directly, e.g. `command_ls` or `command_registry_login`. This function is just an internal bit of bootstrap that attempts to pull the sudo password from the provided json if it was provided, so it shouldn't use that naming scheme. Could just be something like `get_sudo_password`. I'd also say, we probably should just have the rest of
```
if ctx.ssh_user != 'root' and ctx.sudo_json:
      logger.info('Using provided sudo info...')
      command_sudo_login(ctx)    
if ctx.ssh_user != 'root' and ctx.sudo_password:
      logger.info('Using provided sudo password for non root user...')
      cli(['cephadm', 'set-password', ctx.sudo_password])
```
should probably just get pulled into the function as well and it should just become `setup_sudo_password` or something like that. E.g. something like

```
def setup_sudo_password(ctx: cephadmContext) -> None:
    if ctx.ssh_user != 'root' and ctx.sudo_json:
        logger.info('Pulling custom sudo login info from %s.' % ctx.sudo_json)
        d = get_parm(ctx.sudo_json)
        if d.get('password'):
            ctx.sudo_password = d.get('password')
        else:
            raise Error('json provided for custom sudo login did not include "password" field. '
                        'Please setup json file as\n'
                        '{\n'
                        ' "password": "SUDO_PASSWORD"\n'
                        '}')
    if ctx.ssh_user != 'root' and ctx.sudo_password:
        logger.info('Using provided sudo password for non root user...')
        cli(['cephadm', 'set-password', ctx.sudo_password])
```
that would then get called from `prepare_ssh` where the rest of the logic used to live

Reply:
thanks for all the help